src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1593,10 +1598,32 @@ private void HandleCall(MethodDesc callee, MethodSignature signature, ILOpcode o
             {
                 argumentValues[argumentValues.Length - i - 1] = _stack.Pop();
             }
-            PushNonNull(HandleCall(callee, signature, argumentValues, opcode, calliTarget));
+
+            if(constrainedType != null)
+            {
+                if (signature.IsStatic)
+                {
+                    throw new Exception("Constrained call on static method");
+                }
+                StackEntry thisByRef = argumentValues[0];
+                if (thisByRef.Kind != StackValueKind.ByRef)
+                {
+                    throw new Exception("Constrained call without byref");
```

Comment:
At some point (maybe not now), these places should throw a `TypeSystemException.InvalidProgramException` like they do for RyuJIT codegen. We can then have a `catch` for `TypeSystemException` and use `TypeSystemThrowingILEmitter` to generate throwing methods for these (look for how this is used in RyuJIT).

Reply:
Switched the throw to use InvalidProgramExceptions. I think the new method bodies would take some extra refactoring, but we have an approximation of that for WASM that traps (sorta like asserting) if compiling a method threw for any reason.